### ISSUE TITLE
Dataset loader

### ISSUE THREAD

**Issue by pushkarkadam** [pushkarkadam/handpose#2]
- [ ] Dataset class
- [ ] Dataloaders
- [ ] Function that uses both
- [ ] Unit test

**Comment by pushkarkadam** [comment]
Use `plt.imshow(img.permute(1, 2, 0))` for displaying image in pyplot to convert torch tensor to numpy array.

**Comment by pushkarkadam** [comment]
Realised that there could be a problem with using csv file as it does not work for multiple labels in a single image. So, an image gets one row in the csv file and that is used to store the data.

This means that when creating the ground truth tensor, if multiple labels are not taken into account, then the ground truth will not have other objects.

**Comment by pushkarkadam** [comment]
Deciding to switch to the text files based labels

This can separate the multiple labels on different lines and can be treated as two separate labels while creating the ground truth.

**Comment by pushkarkadam** [comment]
Used the following code to conver:

```python
import pandas as pd
import os

df = pd.read_csv('../data/dev/valid.csv')

def save_row_to_txt(row, output_dir):
    
    filename = os.path.join(output_dir, row['image_name'][:-4] + '.txt')
    with open(filename, 'w') as f:
        # Write each column value to the file
        for col in row[1:].index:
            f.write(f"{row[col]} ")

for index, row in df.iterrows():
    save_row_to_txt(row, 'valid_labels')
```

**Comment by pushkarkadam** [comment]
Testing a function that can have multiple labels in a single text file.

Remember filename `00097196_l.txt` in the train labels for dev dataset.

**Comment by pushkarkadam** [comment]
Sorted out the label and image mismatch.

Working on multiple labels per image. 

I think I should work on `label_tensor` function and sort this issue here.

**Comment by pushkarkadam** [comment]
Thought about passing cell coordinates from `lable_tensor` to `truth_head` and eventually to the dataloader. But it messes the tuple up when the dataloader iterates over the batch size and combines all the tensors.

This can be solved by using the `conf` tensor from the `head` and identifying non-zero coordinates.

This can be run through all the tensors in the batch size to gain the coordinates.
